plugins/out_s3/s3.c

```diff
@@ -648,13 +649,25 @@ static int cb_s3_init(struct flb_output_instance *ins,
                           "upload_chunk_size can not be larger than total_file_size");
             return -1;
         }
-        if (ctx->upload_chunk_size < MIN_CHUNKED_UPLOAD_SIZE) {
-            flb_plg_error(ctx->ins, "upload_chunk_size must be at least 5,242,880 bytes");
-            return -1;
-        }
-        if (ctx->upload_chunk_size > MAX_CHUNKED_UPLOAD_SIZE) {
-            flb_plg_error(ctx->ins, "Max upload_chunk_size is 50M");
-            return -1;
+        if (ctx->compression != FLB_AWS_COMPRESS_GZIP) {
+            if (ctx->upload_chunk_size < MIN_CHUNKED_UPLOAD_SIZE) {
+                flb_plg_error(ctx->ins, "upload_chunk_size must be at least 5,242,880 bytes");
+                return -1;
+            }
+            if (ctx->upload_chunk_size > MAX_CHUNKED_UPLOAD_SIZE) {
+                flb_plg_error(ctx->ins, "Max upload_chunk_size is 50MB");
+                return -1;
+            }
+        } else {
+            // todo: remove this hard request and fix the bug complete_multipart_upload break fluent-bit
```

Comment:
I know this is temporary, but Fluent BIt always uses `/* */` style comments

Reply:
removed